sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/RosterBean.java

```diff
@@ -121,50 +125,71 @@ public void init() {
+
 		// Construct the decorated enrollments for the UI
 		decorateEnrollments(siteStudents, sectionEnrollments, assignedSections);
 	}
 
 	private void decorateEnrollments(List<EnrollmentRecord> siteStudents, SectionEnrollments sectionEnrollments, List<CourseSection> assignedSections) {
-		unpagedEnrollments = new ArrayList<EnrollmentDecorator>();
-		for(Iterator<EnrollmentRecord> studentIter = siteStudents.iterator(); studentIter.hasNext();) {
-			EnrollmentRecord enrollment = studentIter.next();
-
+		unpagedEnrollments = new ArrayList<>();
+		siteStudents.forEach(enrollment -> {
 			// Build a map of categories to sections in which the student is enrolled
-			Map<String, CourseSection> map = new HashMap<String, CourseSection>();
-			for(Iterator catIter = categories.iterator(); catIter.hasNext();) {
-				String cat = (String)catIter.next();
-				CourseSection section = sectionEnrollments.getSection(enrollment.getUser().getUserUid(), cat);
-                map.put(cat, section);
-			}
+			Map<String, CourseSection> studentSectionsMap = new HashMap<>();
 
-			// Check to see whether this enrollment should be filtered out
-			boolean includeStudent = false;
-			if(StringUtils.trimToNull(getFilter()) == null) {
-				includeStudent = true;
-			} else {
-				for(Iterator<Entry<String, CourseSection>> entryIter = map.entrySet().iterator(); entryIter.hasNext();) {
-					Entry<String, CourseSection> entry = entryIter.next();
-					CourseSection section = entry.getValue();
-					// Some map entries won't have a section at all, since the student isn't in any section of that category
-					if(section == null) {
-						continue;
-					}
-					if("MY".equals(getFilter()) && assignedSections.contains(section)) {
-						includeStudent = true;
-						break;
-					} else if(section.getCategory().equals(getFilter()) && assignedSections.contains(section)) {
-						includeStudent = true;
-						break;
+			// Iterate site sections to see if the student is a member of and the filter matches the selected section
+			boolean includeStudent = categories.stream().filter(category -> {
+				CourseSection section = sectionEnrollments.getSection(enrollment.getUser().getUserUid(), category);
+				boolean isValidSection = false;
+
+				if (section != null) {
+					if(("MY".equals(getFilter()) || section.getCategory().equals(getFilter()))
+							&& assignedSections.contains(section)) {
+						isValidSection = true;
+
+					} else if (StringUtils.isNotBlank(section.getEid()) && section.getEid().equals(getFilter())) {
+						isValidSection = true;
+
+					} else if (StringUtils.isNotBlank(section.getTitle()) && section.getTitle().equals(getFilter())) {
+						isValidSection = true;
 					}
+
+					studentSectionsMap.put(category, section);
 				}
+
+				return isValidSection;
+			}).count() > 0; // If there are valid sections for the current filter, include the student in the view
```

Comment:
```suggestion
			}).anyMatch(); // If there are valid sections for the current filter, include the student in the view
```

Reply:
I've put anyMatch here as it is way quicker than getting all the stream and then counting it after.